cost/google/idle_vm_recommendations/google_vm_recommendations.pt

```diff
@@ -109,10 +109,63 @@ datasource "ds_google_project" do
   end
 end
 
+script "js_google_project", type: "javascript" do
+  parameters "param_project"
+  result "request"
+  code <<-EOS
+  var filter = "(lifecycleState:ACTIVE)"
+
+  // Apply filter by project only if "param_project" was set.
+  if (param_project.length != 0) {
+    // Since we already have a filter applied this AND is required to apply new filter.
+    filter += " AND ("
+    _.each(param_project, function (id, index) {
+      filter += '(projectId = "' + id + '")'
+
+      // Adds OR in case it isn't last item, means there are more items to be added.
+      if ((index + 1) != param_project.length) {
+        filter += " OR "
+      }
+    })
+    filter += ")"
+  }
+
+  var request = {
+    auth: "auth_google",
+    pagination: "google_pagination",
+    host: "cloudresourcemanager.googleapis.com",
+    path: "/v1/projects/",
+    query_params: {
+      "filter": filter
+    }
+  }
+  EOS
+end
+
 datasource "ds_recommenders" do
```

Comment:
I would maybe rearrange some of the datasources to be more similar to [Rightsize VM Recommender policy template](https://github.com/flexera-public/policy_templates/blob/master/cost/google/rightsize_vm_recommendations/google_rightsize_vm_recommendations.pt), e.g., for this `"ds_recommenders"` datasource I would maybe put after the `"ds_vms_with_cpu_utilization"` datasource and script.

Again this makes it a bit easier to follow the flow of the policy template for someone reading it.

Reply:
@nia-vf1 thank you :), I move some datasources at Idle VM since those made sense to me.